src/sources/generator.rs

```diff
@@ -127,74 +181,141 @@ mod tests {
         }
 
         assert_eq!(rx.try_recv(), Err(mpsc::error::TryRecvError::Closed));
     }
 
     #[tokio::test]
-    async fn limits_count() {
+    async fn round_robin_limits_count() {
         let mut rx = runit(
-            r#"lines = ["one", "two"]
+            r#"format = "round_robin"
+               lines = ["one", "two"]
                count = 5"#,
         )
         .await;
 
-        for _ in 0..10 {
+        for _ in 0..5 {
             assert!(matches!(rx.try_recv(), Ok(_)));
         }
         assert_eq!(rx.try_recv(), Err(mpsc::error::TryRecvError::Closed));
     }
 
     #[tokio::test]
-    async fn adds_sequence() {
+    async fn round_robin_adds_sequence() {
         let message_key = log_schema().message_key();
         let mut rx = runit(
-            r#"lines = ["one", "two"]
-               count = 2
-               sequence = true"#,
+            r#"format = "round_robin"
+               lines = ["one", "two"]
+               sequence = true
+               count = 5"#,
         )
         .await;
 
-        for line in &["1 one", "2 two", "3 one", "4 two"] {
+        for n in 0..5 {
             let event = rx.try_recv().unwrap();
             let log = event.as_log();
             let message = log[&message_key].to_string_lossy();
-            assert_eq!(message, *line);
+            assert!(message.starts_with(&n.to_string()));
         }
 
         assert_eq!(rx.try_recv(), Err(mpsc::error::TryRecvError::Closed));
     }
 
     #[tokio::test]
-    async fn obeys_batch_interval() {
+    async fn round_robin_obeys_interval() {
         let start = Instant::now();
         let mut rx = runit(
-            r#"lines = ["one", "two"]
+            r#"format = "round_robin"
+               lines = ["one", "two"]
                count = 3
-               batch_interval = 1.0"#,
+               interval = 1.0"#,
         )
         .await;
 
-        for _ in 0..6 {
+        for _ in 0..3 {
             assert!(matches!(rx.try_recv(), Ok(_)));
         }
         assert_eq!(rx.try_recv(), Err(mpsc::error::TryRecvError::Closed));
+
         let duration = start.elapsed();
         assert!(duration >= Duration::from_secs(2));
     }
+
+    #[tokio::test]
+    async fn apache_common_generates_output() {
+        let mut rx = runit(
+            r#"format = "apache_common"
+            count = 5"#,
+        )
+        .await;
+
+        for _ in 0..5 {
+            assert!(matches!(rx.try_recv(), Ok(_)));
+        }
+        assert_eq!(rx.try_recv(), Err(mpsc::error::TryRecvError::Closed));
+    }
+
+    #[tokio::test]
+    async fn apache_error_generates_output() {
+        let mut rx = runit(
+            r#"format = "apache_error"
+            count = 5"#,
+        )
+        .await;
+
+        for _ in 0..5 {
+            assert!(matches!(rx.try_recv(), Ok(_)));
+        }
+        assert_eq!(rx.try_recv(), Err(mpsc::error::TryRecvError::Closed));
+    }
+
+    #[tokio::test]
+    async fn syslog_generates_output() {
```

Comment:
Could these do a pattern match on the resulting line, maybe a regex, to ensure they adhere to the appropriate format?

Reply:
Ideally, yes, but that's currently a bit beyond my ken re: Rust regex so I'll need to save this PR for next sprint.